File: tests/tests/fixtures/java_repo/src/main/java/com/example/PaymentController.java
package com.example;

import com.example.service.PaymentService;
import com.example.repository.PaymentRepository;

public class PaymentController {
    private final PaymentService service;

    public PaymentController(PaymentService service) {
        this.service = service;
    }

    public String checkout(Order order) {
        return service.charge(order, order.getAmount());
    }

    public String refund(String orderId) {
        return service.refund(orderId);
    }
}

File: tests/tests/fixtures/python_repo/controllers/payment_controller.py
"""controllers/payment_controller.py — fixture for DocWiki tests."""

from __future__ import annotations

import external_lib as ext  # alias
from models import Order  # alias import test
from services.payment_service import PaymentService  # noqa: F401


class PaymentController:
    """Handles HTTP payment endpoints."""

    def __init__(self, service: PaymentService) -> None:
        self.service = service

    def checkout(self, order: Order) -> dict:
        """Process a checkout request."""
        result = self.service.charge(order, order.amount)
        ext.log("charged")  # unresolved external call
        return {"status": "ok", "result": result}

    def refund(self, order_id: str) -> dict:
        receipt = self.service.refund(order_id)
        return {"refunded": receipt}
